pycontrails/datalib/_leo_utils/search.py

```diff
@@ -0,0 +1,259 @@
+"""Tools for searching for low Earth orbit satellite imagery."""
+
+import dataclasses
+import pathlib
+
+import numpy as np
+import pandas as pd
+
+from pycontrails.core import Flight
+from pycontrails.utils import dependencies
+
+try:
+    import geojson
+except ModuleNotFoundError as exc:
+    dependencies.raise_module_not_found_error(
+        name="datalib._leo_utils module",
+        package_name="geojson",
+        module_not_found_error=exc,
+        pycontrails_optional_package="sat",
+    )
+
+_path_to_static = pathlib.Path(__file__).parent / "static"
+ROI_QUERY_FILENAME = _path_to_static / "bq_roi_query.sql"
+
+
+#: GeoJSON polygon that covers the entire globe.
+GLOBAL_EXTENT = geojson.dumps(
+    geojson.Polygon([[(-180, -90), (180, -90), (180, 90), (-180, 90), (-180, -90)]])
+)
+
+
+@dataclasses.dataclass
+class ROI:
+    """Spatiotemporal region of interest."""
+
+    #: Start time
+    start_time: np.datetime64
+
+    #: End time
+    end_time: np.datetime64
+
+    #: GeoJSON representation of spatial ROI.
+    extent: str
+
+    def __post_init__(self) -> None:
+        """Validate region of interest."""
+        if self.start_time > self.end_time:
+            msg = "start_time must be before end_time"
+            raise ValueError(msg)
+
+        try:
+            decoded = geojson.Feature(geometry=geojson.loads(self.extent))
+        except Exception as exc:
+            msg = "extent cannot be converted to GeoJSON structure"
+            raise ValueError(msg) from exc
+        if not decoded.is_valid:
+            msg = "extent is not valid GeoJSON"
+            raise ValueError(msg)
+
+
+def track_to_geojson(lon: np.ndarray, lat: np.ndarray) -> str:
```

Comment:
Should we reconcile this with https://github.com/contrailcirrus/pycontrails/blob/b08e971f87166710eb41f4b61ba86eab687056f9/pycontrails/core/flight.py#L1351 and https://github.com/contrailcirrus/pycontrails/blob/b08e971f87166710eb41f4b61ba86eab687056f9/pycontrails/core/flight.py#L1376

(Also for reference vector has one too https://github.com/contrailcirrus/pycontrails/blob/b08e971f87166710eb41f4b61ba86eab687056f9/pycontrails/core/vector.py#L1955


Can we abstract those methods in a way that works for this method? or at least use the same approaches in both? Then we can add a "See Also" section to the docstring with 

```
See Also
---------
:meth:`Flight.to_geojson_linestring`
:meth:`Flight.to_geojson_multilinestring`

(We ended up writing our own implementation to avoid the `geojson` dependency - although that dependency looks OK to include even at the core level.)

Reply:
To be honest I didn't realize these methods exist. Yes, we should reconcile them with what I implemented!

In https://github.com/contrailcirrus/pycontrails/pull/200/commits/ce0d49475a123b1006898c7f6f38332c6dc66cc5 I
- modified `Flight.to_geojson_multilinestring` to allow multiple antimeridian crossings and to make the grouping `key` parameter optional, and
- updated `track_to_geojson` to use `Flight.to_geojson_multilinestring` for antimeridian splitting. I still have to do some postprocessing work in `track_to_geojson`, since I don't want altitude coordinates and I want to pad linestrings so they extend all the way to the antimeridian before and after crossings, but this at least means we don't have two independent versions of antimeridian splitting logic to maintain.

This introduces a breaking change, but only in the return type of `flight._antimeridian_index` (was `int`, now is `list[int]`). Let me know what you think!

(Where if anywhere is `Flight.to_geojson_multilinestring` used? As far I can tell it's not used in pycontrails except in tests, and now in `track_to_geojson`.)